aioftp/client.py

```diff
@@ -338,6 +341,153 @@ def parse_directory_response(self, s):
 
         return pathlib.PurePosixPath(directory)
 
+    @staticmethod
+    def parse_unix_mode(s):
+        """
+        Parsing unix mode strings("rwxr-x--t") into
+        hexacimal notation
+
+        :param s: mode string
+        :type s: :py:class:`str`
+
+        :return mode:
+        :rtype: :py:class:`int`
+        """
+        parse_rw = {"rw": 6, "r-": 4, "-w": 2, "--": 0}
+
+        mode = 0
+        mode |= parse_rw[s[0:2]] << 6
+        mode |= parse_rw[s[3:5]] << 3
+        mode |= parse_rw[s[6:8]]
+
+        if s[2] == 's':
+
+            mode |= 0o4100
+
+        elif s[2] == 'x':
+
+            mode |= 0o0100
+
+        elif s[2] != '-':
+
+            raise ValueError
+
+        if s[5] == 's':
+
+            mode |= 0o2010
+
+        elif s[5] == 'x':
+
+            mode |= 0o0010
+
+        elif s[5] != '-':
+
+            raise ValueError
+
+        if s[8] == 't':
+
+            mode |= 0o1000
+
+        elif s[8] == 'x':
+
+            mode |= 0o0001
+
+        elif s[8] != '-':
+
+            raise ValueError
+
+        return mode
+
+    @staticmethod
+    def parse_ls_date(s):
+        """
+        Parsing dates from the ls unix utility
+        ("Nov 18  1958" and "Nov 18 12:29")
+
+        :param s: ls date
+        :type s: :py:class:`str`
+
+        :rtype: :py:class:`str`
+        """
+        try:
+            d = time.strptime(s, "%b %d %H:%M")
+        except ValueError:
+            d = time.strptime(s, "%b %d  %Y")
+        return time.strftime("%Y%m%d%H%M00", d)
+
+    def parse_list_line(self, b):
+        """
+        Attempt to parse a LIST line(similar to unix ls utility)
+
+        :param b: LIST line
+        :type b: :py:class:`bytes` or :py:class:`str`
+
+        :return: (path, info)
+        :rtype: (:py:class:`pathlib.PurePosixPath`, :py:class:`dict`)
+        """
+        if isinstance(b, bytes):
+
+            s = str.rstrip(bytes.decode(b, encoding="utf-8"))
+
+        else:
+
+            s = b
+
+        info = {}
+
+        if s[0] == '-':
+
+            info["type"] = "file"
+
+        elif s[0] == 'd':
+
+            info["type"] = "dir"
+
+        elif s[0] == 'l':
+
+            info["type"] = "link"
+
+        else:
+
+            info["type"] = "unknown"
+
+        # TODO: handle symlinks(beware the symlink loop)
+        info["unix.mode"] = self.parse_unix_mode(s[1:10])
+        s = s[10:].lstrip()
+        i = s.index(' ')
+        info["unix.links"] = s[:i]
+
+        if not info["unix.links"].isdigit():
+
+            raise ValueError
```

Comment:
Do you mean to put an int in the info dictionary? I originally decided against it because all other components are strings and to be consistent with MLSD. I also thought to just do `int(s[:i])` on its own line, but decided against that because it is less readable than `isdigit()`. If you want, I can do that, but clarify which one it is that you mean.